modules/siddhi-core/src/main/java/org/wso2/siddhi/core/util/ExecutionPlanRuntimeBuilder.java

```diff
@@ -88,9 +88,7 @@ public ExecutionPlanRuntimeBuilder(ExecutionPlanContext executionPlanContext) {
     public void defineStream(StreamDefinition streamDefinition) {
         DefinitionParserHelper.validateDefinition(streamDefinition, streamDefinitionMap, tableDefinitionMap,
                 windowDefinitionMap);
-        if (!streamDefinitionMap.containsKey(streamDefinition.getId())) {
-            streamDefinitionMap.putIfAbsent(streamDefinition.getId(), streamDefinition);
-        }
+        streamDefinitionMap.put(streamDefinition.getId(), streamDefinition);
```

Comment:
Sorry, my bad, only the first definition has annotation (the defined one) and others will not, so we should not overwrite the first one. 
Yes equals() consider annotations thats why equalsIgnoreAnnotations() was implemented.